src/libopensc/ctx.c

```diff
@@ -923,6 +955,17 @@ int sc_wait_for_event(sc_context_t *ctx, unsigned int event_mask, sc_reader_t **
 	return SC_ERROR_NOT_SUPPORTED;
 }
 
+#ifdef USE_OPENSSL3_LIBCTX
+void sc_openssl3_deinit(sc_context_t *ctx)
+{
+	if (ctx->ossl3ctx.legacyprov) OSSL_PROVIDER_unload(ctx->ossl3ctx.legacyprov);
+	if (ctx->ossl3ctx.defprov) OSSL_PROVIDER_unload(ctx->ossl3ctx.defprov);
+	if (ctx->ossl3ctx.libctx) OSSL_LIB_CTX_free(ctx->ossl3ctx.libctx);
+	ctx->ossl3ctx.legacyprov = NULL;
+	ctx->ossl3ctx.defprov = NULL;
+	ctx->ossl3ctx.libctx = NULL;
+}
+#endif
```

Comment:
this is also nit, but I would prefer to keep move the deinit function to the same ifdef where the init lives to keep them together. But I understand the idea keeping that close to the "release context" functions.

Reply:
Yeah sounds better to me to keep the code close together, we'd only go down from 4 ifdefs blocks to 3, doesn't look like we'd gain much.